Source/ProxyConnect.swift

```diff
@@ -0,0 +1,428 @@
+        self.url = url;
+        if ["wss", "https"].contains(url.scheme) {
+            self.secure = true
+        }
+    }
+    
+    public func openNetworkStream(completionHandler:(error:NSError?, readStream:NSInputStream?, writeStream : NSOutputStream?) -> Void) {
+        connectDoneHandler = completionHandler
+        configureProxy()
+    }
+    
+    private func didConnect() {
+        ProxyFastLog("_didConnect, return streams");
+        if secure {
+            if httpProxyHost != nil {
+                // Must set the real peer name before turning on SSL
+                ProxyFastLog("proxy set peer name to real host \(url.host)")
+                outputStream?.setProperty(url.host,forKey:"_kCFStreamPropertySocketPeerName")
+            }
+        }
+        
+        receivedHTTPHeaders = nil
+        
+        if let stream = inputStream {
+            CFReadStreamSetDispatchQueue(stream, nil)
+        }
+        inputStream?.delegate = nil
+        outputStream?.delegate = nil
+        if connectDoneHandler != nil {
+            connectDoneHandler!(error:nil, readStream:inputStream, writeStream:outputStream)
+        }
+        
+    }
+    
+    private func connectionFailed(error: NSError?) {
+        ProxyFastLog("_failWithError, return error");
+        var err = error
+        if err == nil  {
+            err  = NSError(domain: "Proxy", code : Int(500),
+                           userInfo : [NSLocalizedDescriptionKey:"Proxy Error"])
+        }
+        
+        receivedHTTPHeaders = nil
+        
+        outputStream?.delegate = nil
+        inputStream?.delegate = nil
+        if let stream = inputStream {
+            CFReadStreamSetDispatchQueue(stream, nil)
+            stream.close()
+        }
+        if let stream = outputStream {
+            CFWriteStreamSetDispatchQueue(stream, nil)
+            stream.close()
+        }
+        outputStream = nil
+        inputStream = nil
+        if connectDoneHandler != nil {
+            connectDoneHandler!(error:err, readStream:nil, writeStream:nil)
+        }
+    }
+    
+    // get proxy setting from device setting
+    private func configureProxy () {
+        ProxyFastLog("configureProxy");
+        var hURL:NSURL? = url;
+        if let host = url.host {
+            if secure {
+                hURL = NSURL(string : "https://"+host)
+            } else {
+                hURL = NSURL(string : "http://"+host)
+            }
+        }
+        if hURL == nil {
+            hURL = url
+        }
+        if let proxySettings: NSDictionary = CFNetworkCopySystemProxySettings()?.takeRetainedValue() {
+            let proxies : NSArray = CFNetworkCopyProxiesForURL(hURL!, proxySettings).takeRetainedValue()
+            if proxies.count == 0 {
+                ProxyFastLog("configureProxy no proxy")
+                openConnection()
+                return
+            }
+            
+            let settings:NSDictionary = proxies[0] as! NSDictionary
```

Comment:
Like here, you don't need to explicitly type since you're doing `as!`. You're also inconsistent with your `:`s. The style used should be `let x: Type`